src/Storages/System/StorageSystemZooKeeperInfo.cpp

```diff
@@ -0,0 +1,392 @@
+        /* 24 */ {"connections", std::make_shared<DataTypeUInt64>(), "The ZooKeeper connections."},
+        /* 25 */ {"outstanding", std::make_shared<DataTypeUInt64>(), "The ZooKeeper outstanding."},
+        /* 26 */ {"zxid", std::make_shared<DataTypeInt64>(), "The ZooKeeper zxid."},
+        /* 27 */ {"node_count", std::make_shared<DataTypeUInt64>(), "The ZooKeeper node count."},
+
+
+        /// dirs command
+        /* 28 */ {"snapshot_dir_size", std::make_shared<DataTypeUInt64>(), "The ZooKeeper snapshot directory size."},
+        /* 29 */ {"log_dir_size", std::make_shared<DataTypeUInt64>(), "The ZooKeeper log directory size."},
+
+
+        /// lgif command
+        /* 30 */ {"first_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper first log index."},
+        /* 31 */ {"first_log_term", std::make_shared<DataTypeUInt64>(), "The ZooKeeper first log term."},
+        /* 32 */ {"last_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last log index."},
+        /* 33 */ {"last_log_term", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last log term."},
+        /* 34 */ {"last_committed_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last committed index."},
+        /* 35 */ {"leader_committed_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper leader committed log index."},
+        /* 36 */ {"target_committed_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper target committed log index."},
+        /* 37 */ {"last_snapshot_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last snapshot index."},
+    };
+}
+
+static std::map<String,String> getTokens(String response, char separator)
+{
+    std::vector<String> result_split;
+    boost::split(result_split, response, [](char c) { return c == '\n'; });
+
+    std::map<String,String> responses_map;
+    for (auto & line : result_split)
+    {
+        if (line.empty())
+            break;
+
+        std::vector <String> line_result;
+
+        boost::split(line_result, line, [separator](char c) { return c == separator; });
+
+        assert(line_result.size() == 2);
+        responses_map[line_result[0]] = line_result[1];
+    }
+    return responses_map;
+}
+
+void StorageSystemZooKeeperInfo::fillData(MutableColumns & res_columns, ContextPtr context,
+                                                    const ActionsDAG::Node *, std::vector<UInt8>) const
+{
+    /// Read all Keeper from config
+    ConfigProcessor config_processor(context->getConfigRef().getString("config-file", "config.xml"));
+
+    /// This will handle a situation when clickhouse is running on the embedded config, but config.d folder is also present.
+    ConfigProcessor::registerEmbeddedConfig("config.xml", "<clickhouse/>");
+    auto clickhouse_config = config_processor.loadConfig();
+
+    String cluster_name = "default";
+
+    Poco::AutoPtr<Poco::XML::NodeList> zookeeper_elements = clickhouse_config.preprocessed_xml->getElementsByTagName("zookeeper");
```

Comment:
I don't think we need parse config manually, we have code for it: https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/ZooKeeper/ZooKeeperArgs.h#L27 We can get it from all the connections: https://github.com/ClickHouse/ClickHouse/blob/fa73a5e5e9aa30731a8923cdd8c9ef6bf3192dc5/src/Common/ZooKeeper/ZooKeeperImpl.h#L239, and we can get all connections with: https://github.com/ClickHouse/ClickHouse/blob/fa73a5e5e9aa30731a8923cdd8c9ef6bf3192dc5/src/Interpreters/Context.h#L1217-L1223.

Reply:
addressed it